core/psh/psh.c

```diff
@@ -1080,14 +1080,12 @@ static int psh_login(void)
 		}
 	}
 
-#ifdef PSH_DEFUSRPWDHASH
 	/* validate against defuser */
 	shadow = crypt(passwd, PSH_DEFUSRPWDHASH);
 	memset(passwd, '\0', maxlen);
 	if (shadow != NULL && strcmp(username, "defuser") == 0 && strcmp(shadow, PSH_DEFUSRPWDHASH) == 0)
 		return 1;
-#endif
```

Comment:
If `PSH_DEFUSRPWDHASH` is empty string  as you set in `Makefile`, `crypt()` returns `NULL`.

Reply:
Yes. There is already a check in psh.c which takes care of that situation.

Reply:
Ok. I didn't notice the checking of this condition.